hilo.rb

```diff
@@ -1,24 +1,35 @@
 #!/usr/bin/env ruby
 puts "Welcome to the Hi-Lo game."
 puts "Guess a random number from 1 to 100 (inclusive)."
-puts "You have 6 guesses."
+puts "You will have a total of 6 guesses."
+puts ""
 
 random_num = rand(1..100)
-puts "Please guess a number."
 
-# only allow integer input
-guess = gets
-while guess.strip != guess.to_i.to_s  
-    puts "Guess a number!"
+# gives the user 6 guesses
+for i in 0..5
+    guesses_left = 6 - i
+    puts "You have #{guesses_left} guesses left."
+    puts "Please guess a number."
+
+    # only allow integer input
     guess = gets
-end
-guess = guess.to_i
+    while guess.strip != guess.to_i.to_s  
```

Comment:
I thought you might want to consider using a set to check against `!guess.strip.match(/[1-100]/)`, but what you have actually works really well since if the input is out of range say `200` it correctly tells the player it is `too high!`.